### ISSUE TITLE
gRPC locked to 1.65.1 while 1.65.5 is installed when not using this repo (bump to 1.69.0 with firebase-ios-sdk 11.9.0)

### ISSUE THREAD

**Issue by mikehardy** [invertase/firestore-ios-sdk-frameworks#116]
There appears to be an ongoing need for manual maintenance on the build script here (in lockstep with manual package/release/tag on the grpc-binary repo) to get new gRPC patches and/or minors when they are released

It seems that to match a typical source install of firestore we should be on 1.65.5 now as cocoapods installs that when not using firestore-ios-sdk-frameworks, but that doesn't exist on grpc-binary repo so it is not currently possible.

I imagine 1.66 or 1.67 (or soon 1.68) of grpc may be the correct transitive dependency of firestore in it's podspec in the future, and that won't exist on grpc-binary either without some intervention there

I opened corresponding / related issue https://github.com/google/grpc-binary/issues/19 on the binary repo as any work to get the current bugfixes (or future features) in grpc ingested here will need work in concert there

**Comment by russellwheatley** [comment]
It was a while ago when Mike and I patched this, but I think you're referring to this: https://github.com/invertase/firestore-ios-sdk-frameworks/blob/main/.github/workflows/scripts/extract-versions.sh#L24

We pinned it because the tag was missing for v1.65.0 I believe.

Reference podspec.json: https://github.com/CocoaPods/Specs/blob/master/Specs/3/1/8/FirebaseFirestoreInternal/11.5.0/FirebaseFirestoreInternal.podspec.json#L96

I think we need the upstream issue to be resolved (i.e. gRPC to be tagged properly, for v1.65.5 )which we need to get Package.swift: https://github.com/invertase/firestore-ios-sdk-frameworks/blob/main/.github/workflows/scripts/extract-urls.sh#L11

**Comment by mikehardy** [comment]
Yep - exactly that - I logged more info in the upstream issue as I agree with you I believe they have to go first by having all necessary / desired tags over there before anything can happen here

**Comment by russellwheatley** [comment]
Adding this to remind myself, this FlutterFire issue: https://github.com/firebase/flutterfire/issues/12960

Ought to be fixed in gRPC >= 1.65.2: https://github.com/firebase/firebase-ios-sdk/issues/13115#issuecomment-2266203884

**Comment by mikehardy** [comment]
@russellwheatley I don't think that compile error applies for library consumers, since they are not compiling gRPC from source so they would never trigger it - it is not triggered when consuming the gRPC binary

That said, any other bugfixes from 1.65.1 through 1.65.5 obviously are not included here since gRPC is pinned to 1.65.1

**Comment by mikehardy** [comment]
1.69 of grpc-binary is out https://github.com/google/grpc-binary/releases

targeting ingestion here in concert with the firebase-ios-sdk 11.9.0 release where firebase-ios-sdk plans to move their dependency and use it, per comment from @paulb777 (thanks Paul!) --> https://github.com/google/grpc-binary/issues/19#issuecomment-2640526428

**Comment by mikehardy** [comment]
@paulb777 as I was in there bumping grpc-binary I realized abseil-cpp-binary may also have a release and it does appear to. I have not bumped it as we adopted 11.9.0 because I didn't realize it was there 😬 - however, I think it may need a change as well?

New release -> https://github.com/google/abseil-cpp-binary/releases/tag/1.2024072200.0

**Comment by AndrewEllen** [comment]
I'm getting the exact same error currently for firebasecore.

[!] CocoaPods could not find compatible versions for pod "FirebaseCore":
  In snapshot (Podfile.lock):
    FirebaseCore (= 11.8.1, ~> 11.8.0)

  In Podfile:
    FirebaseFirestore (from `https://github.com/invertase/firestore-ios-sdk-frameworks.git`, tag `11.8.0`) was resolved to 11.8.0, which depends on
      FirebaseFirestoreBinary (= 11.8.0) was resolved to 11.8.0, which depends on
        FirebaseCore (= 11.8.0)

    firebase_crashlytics (from `.symlinks/plugins/firebase_crashlytics/ios`) was resolved to 4.0.4, which depends on
      Firebase/Crashlytics (= 11.8.0) was resolved to 11.8.0, which depends on
        FirebaseCrashlytics (~> 11.8.0) was resolved to 11.8.0, which depends on
          FirebaseCore (~> 11.8.0)


You have either:
 * changed the constraints of dependency `FirebaseCore` inside your development pod `firebase_crashlytics`.
   You should run `pod update FirebaseCore` to apply changes you've made.





My podfile.lock has

- Firebase/Analytics (11.8.0):
    - Firebase/Core
  - Firebase/Auth (11.8.0):
    - Firebase/CoreOnly
    - FirebaseAuth (~> 11.8.0)
  - Firebase/Core (11.8.0):
    - Firebase/CoreOnly
    - FirebaseAnalytics (~> 11.8.0)
  - Firebase/CoreOnly (11.8.0):
    - FirebaseCore (~> 11.8.0)
  - Firebase/Crashlytics (11.8.0):
    - Firebase/CoreOnly
    - FirebaseCrashlytics (~> 11.8.0)
  - Firebase/Firestore (11.8.0):
    - Firebase/CoreOnly
    - FirebaseFirestore (~> 11.8.0)
  - Firebase/Storage (11.8.0):
    - Firebase/CoreOnly
    - FirebaseStorage (~> 11.8.0)
  - firebase_analytics (11.2.1):
    - Firebase/Analytics (= 11.8.0)
    - firebase_core
    - Flutter
  - firebase_auth (5.1.4):
    - Firebase/Auth (= 11.8.0)
    - firebase_core
    - Flutter
  - firebase_core (3.12.1):
    - Firebase/CoreOnly (= 11.8.0)
    - Flutter
  - firebase_crashlytics (4.0.4):
    - Firebase/Crashlytics (= 11.8.0)
    - firebase_core
    - Flutter
  - firebase_storage (12.1.3):
    - Firebase/Storage (= 11.8.0)
    - firebase_core
    - Flutter

**Comment by mikehardy** [comment]
Apologies, @AndrewEllen your error is not related to the work on this issue at all, it is definitely not the exact same issue as your error contains none of the same keywords or descriptions as my original post. I am going to close this issue.

You may need to make sure that your Flutter firebase iOS version is consistent with the version you are attempting to use from this repository and/or that you have run `pod repo update`
